src/amber/router/router.cr

```diff
@@ -34,24 +34,26 @@ module Amber
         @routes_hash["#{route.controller.downcase}##{route.action.to_s.downcase}"] = route
         add_head(route) if route.verb == :GET
         node
-      rescue Radix::Tree::DuplicateError
-        raise Amber::Exceptions::DuplicateRouteError.new(route)
       end
 
       def add_socket_route(route, handler : WebSockets::Server::Handler)
         @socket_routes.push({path: route, handler: handler})
       end
 
       def route_defined?(request)
-        match_by_request(request).found?
+        ! match(request.method, request.path).nil?
       end
 
       def socket_route_defined?(request)
         @socket_routes.map(&.[:path]).includes?(request.path)
       end
 
       def match_by_request(request)
-        match(request.method, request.path)
+        unless route = match?(request.method, request.path)
+          raise Exceptions::RouteNotFound.new(request)
```

Comment:
this isn't the right thing to do because it causes the server to abort the request, but it's sufficient for testing the routeset code.